packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1528,6 +1528,14 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       if (!this.isRelational("<")) {
         return undefined;
       }
+
+      const oldMaybeInArrowParameters = this.state.maybeInArrowParameters;
+      const oldYieldPos = this.state.yieldPos;
```

Comment:
I added `oldMaybeInArrowParameters` and `oldYieldPos`  at @JLHwung 's suggestion. From looking around, it does seem like these are generally reset together. 

I also wasn't entirely sure whether `oldMaybeInArrowParameters` should be set here, or whether it should be `true` or `false`. I opted to match https://github.com/liamfd/babel/blob/master/packages/babel-parser/src/parser/expression.js#L972, `true` seemed more appropriate.